src/core/bpf/restrict_ifaces/restrict-ifaces.bpf.c

```diff
@@ -0,0 +1,52 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+/* <linux/bpf.h> must precede <bpf/bpf_helpers.h> due to integer types
+ * in bpf helpers signatures.
+ */
+#include <linux/bpf.h>
+#include <bpf/bpf_helpers.h>
+
+const volatile __u8 is_allow_list = 0;
+
+/* Map containing the network interfaces indexes.
+ * The interpretation of the map depends on the value of is_allow_list.
+ */
+struct {
+        __uint(type, BPF_MAP_TYPE_HASH);
+        __type(key, __u32);
+        __type(value, __u8);
+} ifaces_map SEC(".maps");
+
+#define DROP 0
+#define PASS 1
+
+static inline int restrict_network_interfaces_impl(struct __sk_buff *sk) {
+        __u32 zero = 0, ifindex;
+        __u8 *lookup_result;
+
+        ifindex = sk->ifindex;
+        lookup_result = bpf_map_lookup_elem(&ifaces_map, &ifindex);
+        if (is_allow_list) {
+                /* allow-list: let the packet pass if iface in the list */
+                if (lookup_result)
+                        return PASS;
+        } else {
+            /* deny-list: let the packet pass if iface *not* in the list */
+                if (!lookup_result)
+                        return PASS;
+        }
+
+        return DROP;
+}
+
+SEC("cgroup_skb/egress")
+int restrict_network_interfaces_egress(struct __sk_buff *sk) {
+        return restrict_network_interfaces_impl(sk);
+}
+
+SEC("cgroup_skb/ingress")
+int restrict_network_interfaces_ingress(struct __sk_buff *sk) {
+        return restrict_network_interfaces_impl(sk);
+}
```

Comment:
@mauriciovasquezbernal @poettering 

A quick comment regarding names:

1. Program and map name lengths should be less than to 16 characters due to `BPF_OBJ_NAME_LEN` [0] (mind `'\0'` at the end).
The problem with the existing names is that reporting tools, e.g. `bpftool p s` will truncate them to first 15 characters or `'restrict_networ'`. 
2. Let's declare a follow naming convention for BPF programs belonging to systemd? Precisely start the names with`sd_` prefix so users can easily understand what it is? Please see [1]

Hence, wdyt about `"sd_rstr_iface_e"` and `"sd_rstr_iface_i"` ?
 
[0] https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/bpf.h#L488
[1] https://github.com/systemd/systemd/pull/17655#issuecomment-826449111

Reply:
maybe "sd_restrictif_e" and "sd_restrictif_i"?

Reply:
Yeah, makes sense. Fixed!